src/nl/surf/eduhub_rio_mapper/api.clj

```diff
@@ -14,16 +16,31 @@
             [ring.middleware.json :refer [wrap-json-response]])
   (:import java.util.UUID))
 
+(defn extract-resource [uri]
+  (when uri
+    (if (str/starts-with? uri "/job/")
+      (-> uri (subs 5) (str/split #"/" 2) last)             ; Remove action
+      (log/error (str "invalid uri: " uri)))))
+
 (defn wrap-job-enqueuer
   [app enqueue-fn]
-  (fn with-job-enqueuer [req]
+  (fn with-job-enqueuer [{:keys [callback] :as req}]
     (let [{:keys [job] :as res} (app req)]
       (if job
-        (let [token (UUID/randomUUID)]
-          (enqueue-fn (assoc job :token token))
-          (assoc res :body {:token token}))
+        (let [attrs {:token    (UUID/randomUUID)
+                     :callback callback
+                     :resource (extract-resource (:uri req))}]
+          (enqueue-fn (into {} (filter val (merge job attrs))))
```

Comment:
Do we really need this `filter val` bit?  Do we really care if there are pair with `nil` values?  If we do, we should only do the things in `attr` because the `job` is passed in and we don't know about the semantics of it's payload here.  But I think we actually don't care and should keep the `assoc` of the original and just add `:callback` and `:resource` there.

Also maybe use the job namespace for the `:callback` and `:resource` keys to avoid clobbering `job` because we don't know about its content here.

Reply:
I've had a lot of issues with `nil` values, especially in combination with clojure/spec, and apparently Rich Hickey doesn't like it either, so I'm avoiding nil values in maps now.

I'll implement the other suggestions.